lint/rule_target_required_matchers.go

```diff
@@ -0,0 +1,91 @@
+package lint
+
+import (
+	"fmt"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type TargetRequiredMatchersRuleSettings struct {
+	Matchers config.Matchers `yaml:"matchers"`
```

Comment:
Now that the matchers are defined in the config, I don't see that there is a default for job and instance. Was that intended?

Probably okay for there not to be a default, which would make this rule a noop by default. Worth mentioning in the docs that without configuration, this does nothing. Also worth noting what a "legacy" config for job and instance would look like.

Reply:
I added a config example for legacy rules.

And yes it was my intention to have a noop by default.